pkg/cluster/generateconfig.go

```diff
@@ -28,143 +24,72 @@ import (
 	"github.com/openshift/installer/pkg/types/validation"
 	"golang.org/x/crypto/ssh"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/wait"
 
 	"github.com/Azure/ARO-RP/pkg/api"
-	"github.com/Azure/ARO-RP/pkg/bootstraplogging"
-	"github.com/Azure/ARO-RP/pkg/cluster"
-	"github.com/Azure/ARO-RP/pkg/util/azureerrors"
-	"github.com/Azure/ARO-RP/pkg/util/deployment"
+	"github.com/Azure/ARO-RP/pkg/api/validate"
 	"github.com/Azure/ARO-RP/pkg/util/pullsecret"
 	"github.com/Azure/ARO-RP/pkg/util/stringutils"
 	"github.com/Azure/ARO-RP/pkg/util/subnet"
 	"github.com/Azure/ARO-RP/pkg/util/version"
 )
 
-func (m *manager) Create(ctx context.Context) error {
-	var err error
-
-	if m.doc.OpenShiftCluster.Properties.Install == nil {
-		// we don't re-call Dynamic on subsequent entries here.  One reason is
-		// that we would re-check quota *after* we had deployed our VMs, and
-		// could fail with a false positive.
-		timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
-		defer cancel()
-		_ = wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
-			err = m.ocDynamicValidator.Dynamic(ctx)
-			if azureerrors.HasAuthorizationFailedError(err) ||
-				azureerrors.HasLinkedAuthorizationFailedError(err) {
-				m.log.Print(err)
-				return false, nil
-			}
-			return err == nil, err
-		}, timeoutCtx.Done())
-		if err != nil {
-			return err
-		}
-	}
-
-	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
-
-	if m.env.DeploymentMode() != deployment.Development {
-		rp := m.acrtoken.GetRegistryProfile(m.doc.OpenShiftCluster)
-		if rp == nil {
-			// 1. choose a name and establish the intent to create a token with
-			// that name
-			rp = m.acrtoken.NewRegistryProfile(m.doc.OpenShiftCluster)
-
-			m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-				m.acrtoken.PutRegistryProfile(doc.OpenShiftCluster, rp)
-				return nil
-			})
-			if err != nil {
-				return err
-			}
-		}
-
-		if rp.Password == "" {
-			// 2. ensure a token with the chosen name exists, generate a
-			// password for it and store it in the database
-			password, err := m.acrtoken.EnsureTokenAndPassword(ctx, rp)
-			if err != nil {
-				return err
-			}
-
-			rp.Password = api.SecureString(password)
-
-			m.doc, err = m.db.PatchWithLease(ctx, m.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
-				m.acrtoken.PutRegistryProfile(doc.OpenShiftCluster, rp)
-				return nil
-			})
-			if err != nil {
-				return err
-			}
-		}
+func (m *manager) checkQuotas(ctx context.Context) error {
```

Comment:
```suggestion
func (m *manager) dynamicValidate(ctx context.Context) error {
```
?

Reply:
Olga's change (which we think is mis-rebased here) added dynamic validation to Update as well.  Given that dynamic validation is now called on Create and Update, does this architecture where dynamic validate is called as part of the pkg/cluster sequence still make sense?

This PR would be moving to:
Create -> func (m *manager) Install -> runSteps -> checkQuotas()
Update -> call dynamic validator

Which is a slightly odd difference.  Would it make more sense for Update -> func (m *manager) Update -> runSteps -> checkQuotas() as well?

In this case, we might want an explicit switch in checkQuotas where we do all the dynamic validation and only exclude the quota step if SSHKey != nil ?

We probably want to get to the point where we have table-based steps in Update, and where the dynamic validation is shared between Install and Update -- I guess I'm saying, should that be done now?